layers/state_tracker/pipeline_state.cpp

```diff
@@ -1094,3 +1094,43 @@ bool LastBound::IsAnyGraphicsShaderBound() const {
         IsValidShaderBound(ShaderObjectStage::TASK) ||
         IsValidShaderBound(ShaderObjectStage::MESH);
 }
+
+bool LastBound::IsBoundSetCompatible(uint32_t set, const vvl::PipelineLayout &pipeline_layout) const {
+    if ((set >= per_set.size()) || (set >= pipeline_layout.set_compat_ids.size())) {
+        return false;
+    }
+    return (*(per_set[set].compat_id_for_set) == *(pipeline_layout.set_compat_ids[set]));
+}
+
+bool LastBound::IsBoundSetCompatible(uint32_t set, const vvl::ShaderObject &shader_object_state) const {
+    if ((set >= per_set.size()) || (set >= shader_object_state.set_compat_ids.size())) {
+        return false;
+    }
+    return (*(per_set[set].compat_id_for_set) == *(shader_object_state.set_compat_ids[set]));
+};
+
+std::string LastBound::DescribeNonCompatibleSet(uint32_t set, const vvl::PipelineLayout &pipeline_layout) const {
+    std::ostringstream ss;
+    if (set >= per_set.size()) {
+        ss << "The set (" << set << ") is out of bounds for the number of sets bound (" << per_set.size() << ")\n";
+    } else if (set >= pipeline_layout.set_compat_ids.size()) {
+        ss << "The set (" << set << ") is out of bounds for the number sets in the non-compatible VkPipelineLayout ("
```

Comment:
```suggestion
        ss << "The set (" << set << ") is out of bounds for the number sets in the non-compatible VkPipelineLayout ("
```
Maybe just me, but reading this I do not get what "the number sets" refers to : /

Reply:
will try rewording